### PR TITLE
Refactor register endpoints

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This refactors the Quarterdeck and Tenant register endpoints so that the workspace and domain are no longer required, as they will be provided during the onboarding phase. Invite logic is also removed, since invitation accept will now happen after the user has verified their email.

Fixes SC-20635

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

- Need to create "stub" organizations with a unique domain since we won't have the organization/workspace until onboarding is complete
- Ensure that invite logic is removed from the endpoint
- Registration endpoints should accept empty organization/workspace

### Definition of Done

- [x] I have manually tested the change running it locally (having rebuilt all containers) or via unit tests 
- [x] I have added unit and/or integration tests that cover my changes
- [ ] I have added new test fixtures as needed to support added tests
- [ ] I have updated the dependencies list if necessary (including updating yarn.lock and/or go.sum)
- [ ] I have recompiled and included new protocol buffers to reflect changes I made if necessary
- [x] Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x] I have notified the reviewer via Shortcut or Slack that this is ready for review
- [ ] Front-end: Checked sm, md, lg screen resolutions for effective responsiveness
- [ ] Backend-end: Documented service configuration changes or created related devops stories

### Reviewer(s) checklist

- [ ] Front-end: I've reviewed the Figma design and confirmed that changes match the spec.
- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/quarterdeck/auth_test.go [hunk 1]
```diff
@@ -31,155 +31,190 @@ func (s *quarterdeckTestSuite) TestRegister() {
...
-	_, err = s.client.Register(ctx, req)
-	s.CheckError(err, http.StatusBadRequest, "only one field can be set: invite_token, project_id")
-
-	// Test error is returned when organization/domain is missing but project ID is provided
-	req.InviteToken = ""
-	req.Organization = ""
-	req.Domain = ""
-	_, err = s.client.Register(ctx, req)
-	s.CheckError(err, http.StatusBadRequest, "missing required field: organization")
+	s.Run("Happy Path", func() {
```
Comment (pdeziel) [inline pkg/quarterdeck/auth_test.go:37]: Thank you for the subtest tip, it's definitely easier to read this now.

File: pkg/tenant/auth.go [hunk 2]
```diff
@@ -72,71 +68,35 @@ func (s *Server) Register(c *gin.Context) {
...
 	// TODO: use worker queue to limit number of go routines for tasks like this
 	// TODO: test in live integration tests to make sure this works
 	hub := sentrygin.GetHubFromContext(c).Clone()
 	go func() {
+		// TODO: We don't have the name of the user here, so we would have to add it
+		// elsewhere.
 		contact := &sendgrid.Contact{
 			Email: params.Email,
 		}
-		contact.ParseName(params.Name)
```
Comment (pdeziel) [inline pkg/tenant/auth.go:139]: Is this going to be an issue? Do we need to update the contact somewhere else once we have the name?

File: pkg/tenant/db/users.go [hunk 3]
```diff
@@ -21,9 +21,15 @@ func CreateUserResources(ctx context.Context, orgName string, member *Member) (e
...
 		OrgID:           member.OrgID,
-		Name:            slug.Make(orgName),
 		EnvironmentType: "development",
 	}
+
+	if member.Workspace == "" {
+		tenant.Name = slug.Make("dev-" + member.Workspace)
+	} else {
+		tenant.Name = "dev-tenant"
+	}
```
Comment (pdeziel) [inline pkg/tenant/db/users.go:31]: Since we no longer have the organization name on Register, I changed this to use the workspace/domain name.

**Review by bbengfort — APPROVED** [review]
Is it possible for the tenant name to be `dev-tenant`?

**Comment by pdeziel** [comment]
> 

Do you mean instead of using the workspace name?

**Comment by bbengfort** [comment]
@pdeziel I mean since that's in the `else` statement; but workspace name is required right? So the `else` should never trigger?

**Pull request merged**